### PR TITLE
Refactors AVM1 object initialization and DoInitAction.

### PR DESCRIPTION
**Pull request by yurydelendik**
(redo of #2261)

New advance step for DoInitAction is created -- it is initializing scripts and (if needed) executing init blocks.

The AVM1 specific methods were moved from AVM1MovieClip has to be moved (currently to the MovieClip) since during scripts initialization and children construction, AVM1 objects shall not exist.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/shumway/2264)

### CODE REVIEW

File: src/flash/display/DisplayObject.ts [hunk 1]
```diff
@@ -180,6 +180,12 @@ module Shumway.AVMX.AS.flash.display {
     CacheAsBitmap                             = 0x010000,
 
     /**
+     * Indicates whether an AVM1 load event needs to initialize an object when place object
```
Comment (tobytailor) [inline src/flash/display/DisplayObject.ts]: load event?

Comment (yurydelendik) [inline src/flash/display/DisplayObject.ts]: fixed

**Comment by yurydelendik** [comment]
> REFTEST TEST-UNEXPECTED-FAIL | firefox-acid-morph-stas-item2 | swfs/acid/acid-morph.swf | trace

intermittent?

/botio reftest

**Comment by yurydelendik** [comment]
http://swf.codeazur.com.br/#57f65466259a9544d7ce4dd94791f2031ad7ef6b224dd21602d9e642ea4db271

**Comment by tobytailor** [comment]
Review status: :shipit: all files reviewed, all discussions resolved, all commit checks successful.
Reviewed files:
- src/avm1/context.ts @ r1
- src/avm1/flash.d.ts @ r1
- src/avm1/lib/AVM1MovieClip.ts @ r1
- src/avm1/lib/AVM1Utils.ts @ r1
- src/flash/avm1.d.ts @ r1
- src/flash/display/DisplayObject.ts @ r1
- src/flash/display/DisplayObjectContainer.ts @ r1
- src/flash/display/MovieClip.ts @ r1
- src/flash/display/Sprite.ts @ r1
- test/test_manifest_bad.json @ r1
- test/test_manifest_trace.json @ r1

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/shumway/2264)

**Pull request merged**